code/game/jobs/job/support.dm

```diff
@@ -451,6 +451,11 @@
 	backpack_contents = list(
 		/obj/item/videocam/advanced = 1)
 
+/datum/outfit/job/librarian/post_equip(mob/living/carbon/human/H, visualsOnly = FALSE)
+	..()
+	if(H.mind)
+		H.grant_all_babel_languages()
```

Comment:
This should probably be altered to exclude:
- Drone language (the one they speak out loud)
- Xenocommon (the language xenos speak out loud)
- `/datum/language/noise` probably.

- Personally, I would exclude the grey-psionics language.

Reply:
ill look into doing this

Reply:
As a Librarian enthusiast and somewhat of coding amateur I decided to look into this

something like
```
/datum/outfit/job/librarian/post_equip(mob/living/carbon/human/H, visualsOnly = FALSE)
	..()
	if(!H.mind)
		return
	for(var/la in GLOB.all_languages)
		var/datum/language/new_language = GLOB.all_languages[la]
		if(new_language.flags & (HIVEMIND| NOBABEL))
			continue
		H.add_language(la)
```
will add all languages that are not hivemind restricted (Grey Hivemind, Xenomorph Hivemind, Drone Hivemind) but will still leave spoken languages (Xenomorph Common, Drone Speak)

what would be a better idea is to go to `code\modules\mob\language.dm` and add bitflag to each language you would want to exclude
```
/datum/language/drone
	name = "Drone"
	flags = RESTRICTED | NOLIBRARIAN
```

and then ~copypasting~ making your own function based on `grant_all_babel_languages()`
```
/mob/proc/grant_all_librarian_languages()
	for(var/la in GLOB.all_languages)
		var/datum/language/new_language = GLOB.all_languages[la]
		if(new_language.flags & NOLIBRARIAN)
			continue
		languages |= new_language
```

Reply:
done